core/server/apps/subscribers/lib/router.js

```diff
@@ -44,9 +45,13 @@ function honeyPot(req, res, next) {
     next();
 }
 
+function validateUrl(url) {
+    return validator.isEmptyOrURL(url) ? url : '';
+}
+
 function handleSource(req, res, next) {
-    req.body.subscribed_url = req.body.location;
-    req.body.subscribed_referrer = req.body.referrer;
+    req.body.subscribed_url = validateUrl(req.body.location);
```

Comment:
I just pushed an updated version of this commit, which I will hope to resolve the confusion here. I think it is coming from the badly named function 😁 

We're not validating here, we're sanitizing. The big difference being, this isn't a case of validating on save for data integrity where we don't want to save the data if any of it doesn't match the validation rules. Rather, it is a case where we want to sanitize for security purposes, to stop nefarious data being submitted.

Does that make more sense?